src/lib/mdns/Advertiser.h

```diff
@@ -179,34 +179,37 @@ class CommissionAdvertisingParameters : public BaseAdvertisingParams<CommissionA
     {
         if (deviceName.HasValue())
         {
-            strncpy(sDeviceName, deviceName.Value(), min(strlen(deviceName.Value()) + 1, sizeof(sDeviceName)));
-            mDeviceName = Optional<const char *>::Value(static_cast<const char *>(sDeviceName));
+            strncpy(mDeviceName, deviceName.Value(), min(strlen(deviceName.Value()) + 1, sizeof(mDeviceName)));
```

Comment:
nit not for this PR: strncpy will not null-terminate the device and the min usage is odd.

Should use methods in ChipMemString.h instead (which unfortunately are somewhat broken and #6777 fixes.
If that PR does not yet go in, I can do a followup (I imagine we want to replace all strncpy calls).